tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts

```diff
@@ -0,0 +1,124 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+import { PlayYamlRuntime } from '../model/core/operations';
+import { KubernetesResourceState } from '../model/core/states';
+import { KubernetesResources } from '../model/core/types';
+import { expect as playExpect, test } from '../utility/fixtures';
+import { createKindCluster, deleteKindCluster, ensureKindCliInstalled } from '../utility/operations';
+import { waitForPodmanMachineStartup } from '../utility/wait';
+
+const CLUSTER_NAME: string = 'kind-cluster';
+const CLUSTER_CREATION_TIMEOUT: number = 300_000;
+const KIND_NODE: string = `${CLUSTER_NAME}-control-plane`;
+const KUBERNETES_CONTEXT = `kind-${CLUSTER_NAME}`;
+const KUBERNETES_NAMESPACE = 'default';
+const DEPLOYMENT_NAME = 'test-deployment-resource';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const skipKindInstallation = process.env.SKIP_KIND_INSTALL ? process.env.SKIP_KIND_INSTALL : false;
+
+test.beforeAll(async ({ runner, welcomePage, page, navigationBar }) => {
+  test.setTimeout(250000);
+  runner.setVideoAndTraceName('kubernetes-edit-yaml');
+
+  await welcomePage.handleWelcomePage(true);
+  await waitForPodmanMachineStartup(page);
+  if (!skipKindInstallation) {
+    const settingsBar = await navigationBar.openSettings();
+    await settingsBar.cliToolsTab.click();
+
+    await ensureKindCliInstalled(page);
+  }
+  await createKindCluster(page, CLUSTER_NAME, true, CLUSTER_CREATION_TIMEOUT);
+});
+
+test.afterAll(async ({ runner, page }) => {
+  test.setTimeout(90000);
+  try {
+    await deleteKindCluster(page, KIND_NODE, CLUSTER_NAME);
+  } finally {
+    await runner.close();
+  }
+});
+
+test.describe('Kubernetes Edit YAML Feature E2E Test', () => {
+  test.skip(
+    !!process.env.GITHUB_ACTIONS && process.env.RUNNER_OS === 'Linux',
+    'Tests suite should not run on Linux platform',
+  );
+  test('Create a Kubernetes deployment resource', async ({ navigationBar }) => {
+    test.setTimeout(80_000);
+    const podsPage = await navigationBar.openPods();
+    await playExpect(podsPage.heading).toBeVisible();
+    const playYamlPage = await podsPage.openPlayKubeYaml();
+    await playExpect(playYamlPage.heading).toBeVisible();
+    const yamlFilePath = path.resolve(__dirname, '..', '..', 'resources', 'kubernetes', `${DEPLOYMENT_NAME}.yaml`);
+    await playYamlPage.playYaml(yamlFilePath, {
```

Comment:
if the object passed into playYaml can be constant, please do it. It is not repeated now, but constant could be defined in the file.